azure-pipelines.yml

```
@@ -36,38 +36,66 @@ jobs:
     # test when we're not in a unicode locale
       py310-Clocale:
         python.version: '3.10'
+        python.architecture: 'x64'
         TOXENV: py310-test-deps
         TOX_TESTENV_PASSENV: LANG LC_ALL
         LANG: C
         LC_ALL: C
         H5PY_ENFORCE_COVERAGE: yes
-    # Build HDF5 1.10 or 1.12 in the test environment
+    # Build HDF5 1.10, 1.12 or 1.14 in the test environment
       py310-deps-hdf51107:
         python.version: '3.10'
+        python.architecture: 'x64'
         TOXENV: py310-test-deps
         HDF5_VERSION: 1.10.7
         HDF5_DIR: $(HDF5_CACHE_DIR)/$(HDF5_VERSION)
         H5PY_ENFORCE_COVERAGE: yes
       py310-deps-hdf51122:
         python.version: '3.10'
+        python.architecture: 'x64'
         TOXENV: py310-test-deps
         HDF5_VERSION: 1.12.2
         HDF5_DIR: $(HDF5_CACHE_DIR)/$(HDF5_VERSION)
         H5PY_ENFORCE_COVERAGE: yes
       py310-deps-hdf51146:
         python.version: '3.10'
+        python.architecture: 'x64'
         TOXENV: py310-test-deps
         HDF5_VERSION: 1.14.6
         HDF5_DIR: $(HDF5_CACHE_DIR)/$(HDF5_VERSION)
         H5PY_ENFORCE_COVERAGE: yes
     # do mpi tests
       py310-deps-hdf51122-mpi:
         python.version: '3.10'
+        python.architecture: 'x64'
         TOXENV: py310-test-mindeps-mpi
         HDF5_VERSION: 1.12.2
         HDF5_DIR: $(HDF5_CACHE_DIR)/$(HDF5_VERSION)
         HDF5_MPI: ON
         CC: mpicc
+    # Latest python version
+      py313-deps-hdf51146:
+        python.version: '3.13'
+        python.architecture: 'x64'
+        TOXENV: py313-test-deps
+        HDF5_VERSION: 1.14.6
+        HDF5_DIR: $(HDF5_CACHE_DIR)/$(HDF5_VERSION)
+        H5PY_ENFORCE_COVERAGE: yes
+    # Free-threading (noGIL) Python
+      py313t-deps-hdf51146:
+        python.version: '3.13'
+        python.architecture: 'x64-freethreaded'
+
+        # FIXME it was decided to not mark the module as freethreading-compatible
+        # for the time being due to instability concerns.
+        # See matching FIXMEs in pyproject.toml and setup_build.py.
+        PYTHON_GIL: 0
+
+        TOXENV: py313t-test-deps
```

I'd suggest jumping straight to 3.14t. I think all our dependencies support it already and it and given the timing, I don't see us retro-actively adding wheels for 3.13t when anyone interested will already have access to the significantly superior 3.14t
```suggestion
        TOXENV: py314t-test-deps
```

---

I tried that already. Doesn't work on Azure CI yet.